### PR TITLE
Docs: Guide to setting initial page load transaction name

### PR DESCRIPTION
**Pull request by bmorelli25**
Closes elastic/apm-agent-js-base#58

@jahtalab - Here's a rough draft of what I'm thinking for this guide. It's written in more of a "blog post" style vs. "documentation" style. I'd like to hear your thoughts on layout, and have you do a technical edit to ensure I'm using the right terminology. 

Note: I changed up the structure of the RUM docs a little bit to make way for this.

### CODE REVIEW

File: docs/custom-transaction-name.asciidoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+https://www.elastic.co/blog/say-heya-to-the-elastic-search-awards
+https://www.elastic.co/blog/and-the-winner-of-the-elasticon-2018-training-subscription-drawing-is
+
+// Documentation
+https://www.elastic.co/guide/en/elastic-stack/current/index.html
+https://www.elastic.co/guide/en/apm/get-started/current/index.html
+https://www.elastic.co/guide/en/infrastructure/guide/current/index.html
+----
+
+By default, <<transaction-name,`transaction.name`>> is set to `unknown`. This means that when viewing transactions in the Kibana UI, all transactions will be grouped under the common transaction name of `unknown`. Grouping every transaction (especially unrelated ones) into one group doesn't make much sense. To fix this, you can use the API to change the default initial page load transaction name with the <<set-initial-page-load-name,`apm.setInitialPageLoadName()`>> method.
```
Comment (jahtalab) [inline docs/custom-transaction-name.asciidoc]: It's actually `Unknown` with capital `U`, It might have been my own mistake, sorry about that!

**Comment by jahtalab** [comment]
Thanks @bmorelli25! Looks good! Just some minor comments!

File: docs/custom-transaction-name.asciidoc (hunk 1 above)
Comment (bmorelli25) [reply docs/custom-transaction-name.asciidoc]: Gotcha! I wasn't sure of the difference between `setInitialPageLoadName` and `pageLoadTransactionName`. I've swapped them around!

**Comment by bmorelli25** [comment]
Thanks for the feedback @jahtalab - I've made the recommended change, and updated the paragraphs to be on multiple lines.

**Comment by jahtalab** [comment]
Thanks @bmorelli25! I've made some comments!

What do you think about adding the code snippet?

**Comment by bmorelli25** [comment]
Great idea on the code snippet @jahtalab! I added it (along with some callouts explaining the code) to the docs

File: docs/custom-transaction-name.asciidoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+https://www.elastic.co/guide/en/elastic-stack/current/index.html
+https://www.elastic.co/guide/en/apm/get-started/current/index.html
+https://www.elastic.co/guide/en/infrastructure/guide/current/index.html
+----
+
+By default, <<transaction-name,`transaction.name`>> is set to `Unknown`.
+This means that when viewing transactions in the Kibana UI,
+all transactions will be grouped under the common transaction name of `Unknown`.
+Grouping every transaction (especially unrelated ones) into one group doesn't make much sense.
+To fix this, you can use the API to change the default initial page load transaction name with the <<page-load-transaction-name,`pageLoadTransactionName`>> method.
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/custom-transaction-name.asciidoc]: Fixed

File: docs/custom-transaction-name.asciidoc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+A common (and easy) way to set transaction names,
+is to use the <<page-load-transaction-name,`pageLoadTransactionName`>> method to set the transaction name to the page URL.
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/custom-transaction-name.asciidoc]: Fixed

File: docs/custom-transaction-name.asciidoc [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+However, this solution won't be helpful as-is.
+Most URLs include unique names - like blog post titles, or query strings.
+As a result, you'll end up with a large number of different transaction names.
+For this solution to work, you'll need to strip away unique information and find commonality between your URLs.
+Looking at the URLs above, all elastic.co blog posts have `/blog/` in the URL, and all docs have `/guide/` in the URL.
+We can use these paths to set the initial page load transaction names to either `blog` or `guide`.
+This process can then be expanded across your site to ensure common transactions are grouped correctly.
```
[the remark this answers is not in the document]
Comment (bmorelli25) [reply docs/custom-transaction-name.asciidoc]: Added, great idea!

**Comment by jahtalab** [comment]
Thanks @bmorelli25, LGTM 👍

**Pull request merged**